packages/core/admin/admin/src/hooks/useAdminUsers/useAdminUsers.js

```diff
@@ -0,0 +1,51 @@
+import { useQuery, useQueryClient } from 'react-query';
+import { useFetchClient } from '@strapi/helper-plugin';
+import { stringify } from 'qs';
+
+/**
+ * @param {int} [undefined] userId - a User id
+ * @param {object} queryParams - a query params object
+ * @param {object} queryOptions - an object extending the default react-query options
+ */
+
+// eslint-disable-next-line default-param-last
+export function useAdminUsers(userId = '', queryParams = {}, queryOptions = {}) {
+  const QUERY_KEY = ['users', userId, queryParams];
+  const queryString = stringify(queryParams, { encode: false });
+
+  const queryClient = useQueryClient();
+  const { get } = useFetchClient();
+
+  const refetchQueries = async () => {
+    await queryClient.refetchQueries(QUERY_KEY);
+  };
+
+  const { data, isError, isLoading } = useQuery(
+    QUERY_KEY,
+    async () => {
+      const {
+        data: { data },
+      } = await get(`/admin/users/${userId}${queryString ? `?${queryString}` : ''}`);
+
+      return data;
+    },
+    queryOptions
+  );
+
+  // in case a user id is passed, the default value should be an object; otherwise
+  // an array
+  let users = userId ? {} : [];
+
+  if (data) {
+    // .results for a list of users
+    users = data?.results ?? data;
+  }
```

Comment:
nah i think what you've done is the right approach – only return one thing, i'm just wondering if it's clearer to do something like this:

```ts
let users = [];
/**
* We only expect and object when passing a userId because we're only fetching one user
*/
if(userId && data){
 users = [data] 
}
else if(Array.isArray(data?.results)){
	users = data.results
}
```

Then the API return is consistent and `users` always returns an array?

You could then use it like:

```tsx
const [me] = useAdminUsers(myId)
// or
const users = useAdminUsers()
```

wdyt?